File: SEM5_CODE/PYTHON/BASIC/15_prime_check.py
number = int(input("Enter a number: "))  # Read the number
is_prime = True                          # Assume the number is prime

if number < 2:                           # Check numbers smaller than two
    is_prime = False                     # Mark the number as not prime
else:
    for divisor in range(2, number):     # Try possible divisors
        if number % divisor == 0:        # Check exact divisibility
            is_prime = False             # Mark the number as not prime
            break                        # Stop after finding a divisor

if is_prime:                             # Check the final result
    print("Prime number")                # Display prime result
else:
    print("Not a prime number")          # Display non-prime result

File: SEM3_CODE/PHP/BASIC/15_prime_check.php
<?php
$number = (int) readline('Enter a number: ');             // Read the number
$isPrime = true;                                          // Assume the number is prime

if ($number < 2) {                                        // Check numbers smaller than two
    $isPrime = false;                                     // Mark the number as not prime
} else {
    for ($divisor = 2; $divisor < $number; $divisor++) {  // Try possible divisors
        if ($number % $divisor == 0) {                    // Check exact divisibility
            $isPrime = false;                             // Mark the number as not prime
            break;                                        // Stop after finding a divisor
        }
    }
}

if ($isPrime) {                                           // Check the final result
    echo 'Prime number';                                  // Display prime result
} else {
    echo 'Not a prime number';                            // Display non-prime result
}
?>
